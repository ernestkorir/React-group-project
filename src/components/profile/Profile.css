.reserved_missions {
    width: 100%;
    max-width: 800px;
    margin: 0 auto;
  }
  
  .missions_table {
    width: 50%;
    border-collapse: collapse;
    margin-top: 20px;
  }
  
  .missions_table th,
  .missions_table td {
    border: 1px solid #ccc;
    padding: 8px;
    text-align: left;
  }
  
  .missions_table th {
    background-color: #f2f2f2;
  }
  
  .missions_table tr:nth-child(even) {
    background-color: #f2f2f2;
  }
  
  .missions_table tr:hover {
    background-color: #ddd;
  }
  