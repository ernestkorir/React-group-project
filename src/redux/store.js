import { configureStore } from '@reduxjs/toolkit';
import rocketReducer from './rockets/rocketsSlice';

const store = configureStore({
  reducer: {
    rocket: rocketReducer,
  },
});

export default store;
