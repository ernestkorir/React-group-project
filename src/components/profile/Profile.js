import { useSelector } from 'react-redux';
import './Profile.css';

const Profile = () => {
  const rocketList = useSelector((state) => state.rocket);
  const rocketFiltered = rocketList.rocket.filter((rocket) => rocket.reserved === true);
  const allMissions = useSelector((state) => state.mission.missions);
  const joinedMissions = allMissions.filter((mission) => mission.reserved === true);

  return (
    <section className="profilePage">
        <div className='mission-reserve'>
          <h2>My Missions</h2>
          <div className='tmission'>
            <ul>
              { joinedMissions.length > 0 ? (
                joinedMissions.map((joined) => (
                <li key={joined.mission_id}>
                  {joined.mission_name}
                </li>
                ))
              ) : (
                <p>No Mission Joined</p>
              )}
            </ul>
          </div>
      </div>
      <div className='rocket-reserve'>
        <h2> My Rockects</h2>
        <div className='trocket'>
          <ul className="display me">
            { rocketFiltered.length > 0 ? (
              rocketFiltered.map((rocket) => (
                <li key={rocket.id}>

                    {rocket.rocketName}

                </li>
              ))
            ) : (
                <p className="title-three">No Rocket booked </p>
            )}
          </ul>
        </div>
      </div>
    </section>
  );
};
export default Profile;
