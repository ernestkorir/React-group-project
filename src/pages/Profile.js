import Profile from '../components/profile/Profile';

const ProfilePage = () => (
  <div>
    <Profile />
  </div>
);
export default ProfilePage;
