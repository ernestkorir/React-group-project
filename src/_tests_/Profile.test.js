import React from "react";
import renderer from "react-test-renderer";
import Profile from "../components/profile/Profile";
import { Provider } from "react-redux";
import store from "../redux/store";

describe("Test Profile component", () => {
  it("Renders well", () => {
    const tree = renderer.create(
    <Provider store={store}> 
        <Profile />
    </Provider>
    )
    .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
