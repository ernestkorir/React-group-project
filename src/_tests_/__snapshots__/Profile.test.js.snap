// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Profile component Renders well 1`] = `
<section
  className="profilePage"
>
  <div
    className="mission-reserve"
  >
    <h2>
      My Missions
    </h2>
    <table
      className="tmission"
    >
      <tbody>
        <p>
          No Mission Joined
        </p>
      </tbody>
    </table>
  </div>
  <div
    className="rocket-reserve"
  >
    <h2>
       My Rockects
    </h2>
    <table
      className="trocket"
    >
      <tbody
        className="display me"
      >
        <p
          className="title-three"
        >
          No Rocket booked 
        </p>
      </tbody>
    </table>
  </div>
</section>
`;
